### PR TITLE
LTD-5002-denial-matches-not-showing

### PR DESCRIPTION
**Pull request by depsiatwal**
https://uktrade.atlassian.net/browse/LTD-5002

This fixes an issue which relates to  unable to see end_user (denial matches) in the make recommendation tab. 
It was originally an intermittent issue. 

Turns out this is now permanently broken as a result of the denial changes.  The Link on the make recommendation page uses reference and if this is no reference then the link is invisible to the user. 

The fix here is to include the new unique reference regime_reg_ref field and use this to display a hyperlink in the UI. 

I have added tests in the UI also to check this is displayed as expected. 

API change is here 
https://github.com/uktrade/lite-api/pull/2073

These can be merged independently with no impact but the change will only work when both PRs have been merged.

### CODE REVIEW

File: unit_tests/caseworker/advice/views/test_select_advice_view.py [hunk 1]
```diff
@@ -55,3 +53,25 @@ def test_serial_numbers_not_added_for_firearm_product_in_select_advice_view(auth
     assert "1. 12345" not in str(product_details)
     assert "2. ABC-123" not in str(product_details)
     assert "Not added yet" in str(product_details)
+
+
+def test_select_advice_get_check_denials(authorized_client, data_standard_case, url):
```
[the remark this answers is not in the document]
Comment (depsiatwal) [reply unit_tests/caseworker/advice/views/test_select_advice_view.py:58]: no done't look like it so was a good spot may have prevented this bug in the first place.

**Review by currycoder — COMMENTED** [review]
Added a comment inline.  Could it also be a good idea to cover this with an E2E test?  Just thinking what would help us catch this more immediately if we come back and change denials API formats in the future.

File: caseworker/advice/templates/advice/case_detail.html [hunk 2]
```diff
@@ -190,31 +190,32 @@ <h3>{{ forloop.counter }}. {{ good.good.name }}</h3>
...
+          <div class="govuk-summary-list__row" id="denial-matches">
+              <dt class="govuk-summary-list__key">
+                Denial matches
+              </dt>
+              <dd class="govuk-summary-list__value">
+                  {% if case.data.denial_matches %}
+                    {% for denial in case.data.denial_matches %}
+                      <p>
+                        <a href="{% url 'external_data:denial-detail' pk=denial.denial_entity.id %}?queue_id={{queue.id}}&case_id={{case.id}}">
+                          {{ denial.denial_entity.regime_reg_ref }}
```
[the remark this answers is not in the document]
Comment (depsiatwal) [reply caseworker/advice/templates/advice/case_detail.html]: repushed and try to minimise the indenting

File: unit_tests/caseworker/advice/views/test_select_advice_view.py [hunk 3]
```diff
@@ -55,3 +53,25 @@ def test_serial_numbers_not_added_for_firearm_product_in_select_advice_view(auth
...
+    denial_match_1 = data_standard_case["case"]["data"]["denial_matches"][0]["denial_entity"]
+    denial_match_2 = data_standard_case["case"]["data"]["denial_matches"][1]["denial_entity"]
+    response = authorized_client.get(url)
+    assert response.status_code == 200
+
+    soup = BeautifulSoup(response.content, "html.parser")
+    search_input = soup.find("div", id="denial-matches")
+
+    denial_1 = search_input.find_all("p")[0]
```
[the remark this answers is not in the document]
Comment (depsiatwal) [reply unit_tests/caseworker/advice/views/test_select_advice_view.py]: added a class name

**Comment by depsiatwal** [comment]
> What do you think about the E2E test idea?

Yeah I agree think it would be useful just seeing if there any natural place I can slot in into our e2e tests

**Review by saruniitr — APPROVED** [review]

**Pull request merged**